server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserMailboxesRoutesTest.java

```diff
@@ -202,6 +204,84 @@ void putShouldReturnNotFoundWithNonExistingUser() throws Exception {
                 .containsEntry("message", "Invalid get on user mailboxes");
         }
 
+        @Test
+        void putShouldThrowWhenMailBoxNameWithDots() throws Exception {
+            Map<String, Object> errors = when()
+                .put(MAILBOX_NAME_WITH_DOTS)
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .jsonPath()
+                .getMap(".");
+
+            assertThat(errors)
+                .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+                .containsEntry("type", "InvalidArgument")
+                .containsEntry("message", "Attempt to create an invalid mailbox")
+                .containsEntry("details", "'#private:username:my..MailboxName' has an empty mailbox name within its hierarchy");
+        }
+
+        @Test
+        void putShouldThrowWhenMailBoxNameStartWithDot() throws Exception {
+            Map<String, Object> errors = when()
+                .put(".startWithDot")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .jsonPath()
+                .getMap(".");
+
+            assertThat(errors)
+                .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+                .containsEntry("type", "InvalidArgument")
+                .containsEntry("message", "Attempt to create an invalid mailbox")
+                .containsEntry("details", "'#private:username:.startWithDot' has an empty mailbox name within its hierarchy");
+        }
+
+        @Test
+        void putShouldThrowWhenMailBoxNameEndWithDots() throws Exception {
+            Map<String, Object> errors = when()
+                .put("endWithDot.")
+            .then()
+                .statusCode(HttpStatus.BAD_REQUEST_400)
+                .contentType(ContentType.JSON)
+                .extract()
+                .body()
+                .jsonPath()
+                .getMap(".");
+
+            assertThat(errors)
+                .containsEntry("statusCode", HttpStatus.BAD_REQUEST_400)
+                .containsEntry("type", "InvalidArgument")
+                .containsEntry("message", "Attempt to create an invalid mailbox")
+                .containsEntry("details", "'#private:username:endWithDot.' has an empty mailbox name within its hierarchy");
```

Comment:
maybe we should go a little further in these explanations and explain that the hierarchy separator is `.`? I guess it could be non obvious for users nor admins